@pohly One of the test is saying that 
`# k8s.io/kubernetes/test/e2e/framework
test/e2e/framework/util.go:1381:9: Failf call has arguments but no formatting directives
# k8s.io/kubernetes/test/e2e/framework/providers/kubemark
test/e2e/framework/providers/kubemark/kubemark.go:79:19: Failf call has arguments but no formatting directives`
**so, should I add %v format directive in the Failf function.**

--- a/test/e2e/framework/providers/kubemark/kubemark.go
+++ b/test/e2e/framework/providers/kubemark/kubemark.go
@@ -75,7 +75,9 @@ func (p *Provider) FrameworkBeforeEach(f *framework.Framework) {
 		p.controller, err = kubemark.NewKubemarkController(externalClient, externalInformerFactory, f.ClientSet, kubemarkNodeInformer)
 		framework.ExpectNoError(err)
 		externalInformerFactory.Start(p.closeChannel)
-		framework.ExpectEqual(p.controller.WaitForCacheSync(p.closeChannel), true)
+		if !p.controller.WaitForCacheSync(p.closeChannel) {
+			framework.Failf("Unable to sync caches for %v", p.controller)
+		}
 		go p.controller.Run(p.closeChannel)
 	}
 }

--- a/test/e2e/framework/util.go
+++ b/test/e2e/framework/util.go
@@ -1377,7 +1377,7 @@ retriesLoop:
 		// NOTE the test may need access to the events to see what's going on, such as a change in status
 		actualWatchEvents := scenario(resourceWatch)
 		errs := sets.NewString()
-		ExpectEqual(len(expectedWatchEvents) <= len(actualWatchEvents), true, "Error: actual watch events amount (%d) must be greater than or equal to expected watch events amount (%d)", len(actualWatchEvents), len(expectedWatchEvents))
+		gomega.Expect(len(expectedWatchEvents)).To(gomega.BeNumerically("<=", len(actualWatchEvents)), "Did not get enough watch events")
 
 		totalValidWatchEvents := 0
 		foundEventIndexes := map[int]*int{}
@@ -1406,7 +1406,9 @@ retriesLoop:
 			fmt.Println("invariants violated:\n", strings.Join(errs.List(), "\n - "))
 			continue retriesLoop
 		}
-		ExpectEqual(errs.Len() > 0, false, strings.Join(errs.List(), "\n - "))
+		if errs.Len() > 0 {
+			Failf("Unexpected error(s): %v", strings.Join(errs.List(), "\n - "))
+		}
 		ExpectEqual(totalValidWatchEvents, len(expectedWatchEvents), "Error: there must be an equal amount of total valid watch events (%d) and expected watch events (%d)", totalValidWatchEvents, len(expectedWatchEvents))
 		break retriesLoop
 	}

